apps/main/src/api/borrow/hooks.ts

```diff
@@ -0,0 +1,247 @@
+import { zip } from "remeda"
+
+import { useBorrowReserves } from "@/api/borrow/queries"
+import { useDefillamaLatestApyQueries } from "@/api/external/defillama"
+import { isStableSwap, useAssets } from "@/providers/assetsProvider"
+
+type UnderlyingAssetApy = {
+  id: string
+  isStaked: boolean
+  supplyApy: number
+  borrowApy: number
+}
+
+export type ApyType = "supply" | "borrow"
+export type BorrowAssetApyData = {
+  assetId: string
+  tvl: string
+  vDotApy?: number
+  lpAPY: number
+  incentivesNetAPR: number
+  incentives: ReserveIncentiveResponse[]
+  underlyingAssetsApyData: UnderlyingAssetApy[]
+  underlyingSupplyApy: number
+  underlyingBorrowApy: number
+  totalSupplyApy: number
+  totalBorrowApy: number
+  // farms: TFarmAprData[] | undefined @TODO farms
+  // stablepoolData: TStablePoolDetails | undefined @TODO stablepoolData
+}
+
+export const useBorrowAssetsApy = (assetIds: string[]) => {
+  const { getAsset, getErc20AToken } = useAssets()
+  const { data: borrowReserves, isLoading } = useBorrowReserves()
+
+  const assetIdsMemo = useStableArray(assetIds)
+
+  const reserves = useMemo(
+    () => borrowReserves?.formattedReserves ?? [],
+    [borrowReserves],
+  )
+
+  const allAssetIds = useMemo(() => {
+    const ids = assetIdsMemo.flatMap((assetId) => {
+      const asset = getAsset(assetId)
+      return asset && isStableSwap(asset) && asset.underlyingAssetId
+        ? asset.underlyingAssetId
+        : [assetId]
+    })
+    return [...new Set(ids)]
+  }, [assetIdsMemo, getAsset])
+
+  const externalApys = useDefillamaLatestApyQueries(allAssetIds)
+  const externalApysMap = useMemo(() => {
+    const externalApyEntries = zip(allAssetIds, externalApys)
+    return new Map(externalApyEntries)
+  }, [allAssetIds, externalApys])
+
+  const data = useMemo<BorrowAssetApyData[]>(() => {
+    if (isLoading) return []
+    return assetIdsMemo.map((assetId) => {
+      const asset = getAsset(assetId)
+      const assetReserve = reserves.find(
+        ({ underlyingAsset }) =>
+          underlyingAsset === getAddressFromAssetId(assetId),
+      )
+
+      const incentives = assetReserve?.aIncentivesData ?? []
+
+      const stableSwapAssetIds =
+        asset && isStableSwap(asset) && asset.underlyingAssetId
+          ? asset.underlyingAssetId
+          : [assetId]
+
+      const underlyingAssetIds = stableSwapAssetIds.map((assetId) => {
+        return getErc20AToken(assetId)?.underlyingAssetId ?? assetId
+      })
+
+      const underlyingReserves = reserves.filter((reserve) => {
+        return underlyingAssetIds
+          .map(getAddressFromAssetId)
+          .includes(reserve.underlyingAsset)
+      })
+
+      const calculatedData = calculateAssetApyTotals(
+        stableSwapAssetIds,
+        underlyingReserves,
+        assetReserve?.aIncentivesData ?? [],
+        externalApysMap,
+        // getRelatedAToken,
+      )
+
+      return {
+        assetId,
+        tvl: assetReserve?.totalLiquidityUSD || "0",
```

Comment:
Is zero fallback valid for tvl? I wouldn't return any data if assetReserve is undefined, if it's invalid case